src/Couchbase.Lite.Shared/Serialization/FleeceMutableArray.cs

```diff
@@ -152,6 +155,15 @@ public void Set(int index, object val)
 
         #endregion
 
+        #region Internal Method(s)
+
+        internal void NewFLMutableArray()
```

Comment:
What is the point of having this as a separate method?  It seems dangerous because  new object won't be usable until this is called.

Reply:
Because a new FleeceMutableArray object will be created when an ArrayObject is created, but we don't want to call Native.FLMutableArray_New() and set _releaseRequired = true until ArrayObject() in constructed. And I want to do _threadSafety.DoLocked with Native.FLMutableArray_New()..

Reply:
You can do something like this inside of the `ArrayObject` constructor:

`_array = _threadSafety.DoLocked(() => new FLMutableArray())`

Reply:
_array is FleeceMutableArray, and I already have internal readonly FleeceMutableArray _array = new FleeceMutableArray(); as variable in ArrayObject.

Reply:
So move the `new FleeceMutableArray()` to the constructor?  I don't see the benefit of having a separate method just to initialize an object